File: Some_Important_Topics/Factorial.js
// Find factorial with recursion for limited inputed limit

var limit = 15;

const Factorial=(limit)=>{
    if(limit<=1){
        return 1;
    }
    return limit*Factorial(limit-1);
}

const result = Factorial(limit);
console.log(`Final Result : ${result}`); // Final Result : 1307674368000



// same concept but output is diffect for C99 and This code !
// The issue might be related to the limitations of recursion depth and numerical precision in JavaScript

File: Some_Important_Topics/Factorial.c
// Find factorial with recursion for limited inputed limit

#include <stdio.h>

int Factorial(int limit);

int main()
{
    int limit;
    printf("Enter the limit : ");
    scanf("%d",&limit); // Input = 15
    
    int result = Factorial(limit);
    printf("Final Result : %d\n",result); // Final Result : 2004310016
    return 0;
}

int Factorial (int limit){
    if(limit<=1){
        return 1;
    }
    return limit * Factorial(limit-1);
}
